src/main/java/school/lemon/changerequest/java/introduction/hw2/Task9.java

```diff
@@ -1,7 +1,43 @@
 package school.lemon.changerequest.java.introduction.hw2;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
 public class Task9 {
     public static String[] extractData(String URL) {
-        return new String[]{"", "", ""};
+
+        String result[] = new String[]{"", "", ""};
+        if (URL == null)
+            return result;
+
+
+        Pattern p = Pattern.compile("^[a-zA-Z]+:/|^[a-zA-Z]+//");
+        Matcher m = p.matcher(URL);
+
+        if (m.find())
+            result[0] = m.group().substring(0, m.group().length() - 2);
+
+        StringBuffer tmp;
+        if (result[0].length() == 0)
+            p = Pattern.compile("^[a-zA-Z0-9._-]+[/:]?");//|^[a-zA-Z0-9._-]+");
+        else
+            p = Pattern.compile("/[a-zA-Z0-9._-]+[/:]|/[a-zA-Z0-9._-]+");
+        m = p.matcher(URL);
+        if (m.find()) {
+            tmp = new StringBuffer(m.group());
+            p = Pattern.compile("[a-zA-Z0-9._-]+");
+            m = p.matcher(tmp.toString());
+            m.find();
+            result[1] = m.group();
+        }
+
+        p = Pattern.compile(":[0-9]+");
+        m = p.matcher(URL);
+        if (m.find()) {
+            result[2] = m.group().substring(1, m.group().length());
+        }
+
+        return result;
```

Comment:
What about capturing the first part of the URL?

I'd use such regexp: `(?:(\\w+)://)?` - it says, that we have one non-captured group that contain some number of `word`-symbols and `://` at the end - it's general definition of protocol schema. The last `?` says, that this part is optional.

Could you add 2 additional regexp patterns to extract host and port?